src/local-mode.ts

```diff
@@ -56,42 +90,172 @@ export class LocalModeManager {
 
 		if (!localMode) {
 			// Remove any leftovers if necessary
-			await this.removeLocalModeArtifacts();
+			await this.handleLocalModeStateChange(false);
 		}
 	}
 
-	public async removeLocalModeArtifacts(): Promise<void> {
-		try {
-			const images = await this.getLocalModeImages();
-			const containers = await this.getLocalModeContainers(images);
+	// Query the engine to get currently running containers and installed images.
+	public async collectEngineSnapshot(): Promise<EngineSnapshotRecord> {
+		const containers = this.docker
+			.listContainers()
+			.then(resp => _.map(resp, 'Id'));
+		const images = this.docker.listImages().then(resp => _.map(resp, 'Id'));
+		const volumes = this.docker
+			.listVolumes()
+			.then(resp => _.map(resp.Volumes, 'Name'));
+		const networks = this.docker.listNetworks().then(resp => _.map(resp, 'Id'));
```

Comment:
Can you change these names slightly to make it more clear that their intended to be promises, I initially thought you'd dropped the `await` accidentally.